### ISSUE TITLE
LRS returns "Request header field is missing ':' separator" when using long username/passwords

### ISSUE THREAD

**Issue by ajblum** [RusticiSoftware/TinCanPython#85]
When using the main.py test in TinCanPython with LearningLocker's LRS (https://github.com/LearningLocker/learninglocker), the LRS returns:

<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
Request header field is missing ':' separator.<br />

Looking at the remote_lrs.py I find:

            auth_string = "Basic " + base64.encodestring(unicode(kwargs["username"]) +
                                                         ":" +
                                                         unicode(kwargs["password"]))[:-1]

base64.encodestring resulted in newlines added at the end and every 76 characters, so the header was being malformed by having the tail portion of the base64 encoded authentication string on a newline.

My suggestion is to use this instead:

            auth_string = "Basic " + base64.b64encode(unicode(kwargs["username"]) +
                                                         ":" +
                                                          unicode(kwargs["password"]))

Note: changed to use b64encode and remove [:-1].

**Comment by brianjmiller** [comment]
@ajblum seems reasonable, would you like to submit a pull request?

**Comment by jpablo128** [comment]
I think this is similar (or the same) thing I experienced... I did submit a pull request (with a different solution, though)

**Comment by ajblum** [comment]
@brianjmiller sure.
I've got a local branch with that change that I can push.  I don't have permissions to push though:
```
[ablum@badger TinCanPython]$ git push --set-upstream origin ajblum-remote_lrs-auth_string
remote: Permission to RusticiSoftware/TinCanPython.git denied to ajblum.
```
Regards,

Andrew

**Comment by brianjmiller** [comment]
This should be fixed in 0.0.5 based on #88.
